internal/backend/sessions/sessionworkflows/callbacks.go

```
@@ -0,0 +1,159 @@
+)
+
+func (w *sessionWorkflow) start(ctx context.Context, rid sdktypes.RunID, loc sdktypes.CodeLocation, inputs map[string]sdktypes.Value, memo map[string]string) (sdktypes.SessionID, error) {
+	l := w.l.With(zap.Any("rid", rid), zap.Any("loc", loc), zap.Any("inputs", inputs), zap.Any("memo", memo))
+
+	l.Info("session start requested")
+
+	session := sdktypes.NewSession(w.data.Build.ID(), loc, inputs, memo).
+		WithParentSessionID(w.data.Session.ID()).
+		WithDeploymentID(w.data.Session.DeploymentID()).
+		WithProjectID(w.data.Session.ProjectID())
+
+	sid, err := w.ws.sessions.Start(authcontext.SetAuthnSystemUser(ctx), session)
+	if err != nil {
+		return sdktypes.InvalidSessionID, err
+	}
+
+	w.l.Info("session started", zap.Any("sid", sid))
+
+	return sid, nil
+}
+
+func (w *sessionWorkflow) subscribe(ctx context.Context, _ sdktypes.RunID, name, filter string) (string, error) {
+	_, connection := kittehs.FindFirst(w.data.Connections, func(c sdktypes.Connection) bool {
+		return c.Name().String() == name
+	})
+
+	_, trigger := kittehs.FindFirst(w.data.Triggers, func(t sdktypes.Trigger) bool {
+		return t.Name().String() == name
+	})
+
+	if connection.IsValid() && trigger.IsValid() {
+		return "", errors.New("ambiguous source name - matching both a connection and a trigger")
+	}
+
+	var did sdktypes.EventDestinationID
+
+	switch {
+	case connection.IsValid():
+		did = sdktypes.NewEventDestinationID(connection.ID())
+	case trigger.IsValid():
+		did = sdktypes.NewEventDestinationID(trigger.ID())
+	default:
+		return "", fmt.Errorf("source %q not found", name)
+	}
+
+	signalID, err := w.createEventSubscription(sessioncontext.GetWorkflowContext(ctx), filter, did)
+	if err != nil {
+		return "", err
+	}
+
+	return signalID.String(), nil
+}
+
+func (w *sessionWorkflow) unsubscribe(ctx context.Context, rid sdktypes.RunID, signalID string) error {
+	uuid, err := uuid.Parse(signalID)
+	if err != nil {
+		return err
+	}
+
+	w.removeEventSubscription(ctx, uuid)
```

We does `removeEventSubscritpion` does hide the underlying error?

---

This is the same code as before, I just moved it around. @efiShtain ?